src/test.ts

```diff
@@ -0,0 +1,163 @@
+  const packageTarball = path.join(targetDir, 'module-under-test.tgz');
+  const files = await packlist({path: packageDir});
+  await tar.create(
+      {
+        prefix: 'package/',
+        cwd: packageDir,
+        file: packageTarball,
+        gzip: true
+        // tslint:disable-next-line:no-any @types/tar is missing types for
+        // prefix in CreateOptions. TODO: submit fix to @types/tar.
+      } as any,
+      files);
+  return packageTarball;
+}
+
+export async function test(options: TestOptions) {
+  const deps = {...DEFAULT_DEPENDENCIES, ...options.deps};
+  const moduleUnderTest = options.packageDir || process.cwd();
+  let output = '';
+  const installDir = await deps.tmpDir();
+
+  try {
+    const tarball = await pack(moduleUnderTest, installDir);
+    await installInTarget(tarball);
+    await (
+        isTSCodeSample(options.sample) ? checkTS(installDir, options.sample) :
+                                         checkJS(installDir, options.sample));
+  } catch (err) {
+    err.output = output;
+    throw err;
+  } finally {
+    deps.rimraf(installDir);
+  }
+  return;
+
+  function log(message: string) {
+    output += message;
+  }
+
+  async function run(cmd: string, args: string[], options?: SpawnOptions) {
+    return deps.spawn(cmd, args, options, log);
+  }
+
+  async function installInTarget(tarball: string) {
+    await run('npm', ['init', '-y'], {cwd: installDir});
+    await run(
+        'npm', ['install', '--prefer-offline', tarball], {cwd: installDir});
+  }
+
+  async function checkTS(installDir: string, sample: TSCodeSample) {
+    await writeFileP(path.join(installDir, INDEX_TS), sample.ts, 'utf-8');
+
+    if (sample.dependencies) {
+      await run(
+          'npm',
+          ['install', '--prefer-offline', '--save'].concat(sample.dependencies),
+          {cwd: installDir});
+    }
+
+    const devDeps = (sample.devDependencies || []).concat(['typescript@3.x']);
+    await run(
+        'npm', ['install', '--prefer-offline', '--save-dev'].concat(devDeps),
+        {cwd: installDir});
+
+    await run(
+        'node_modules/.bin/tsc', ['--strict', INDEX_TS], {cwd: installDir});
+    await run('node', [path.join(installDir, INDEX_JS)], {cwd: installDir});
+  }
+
+  async function checkJS(installDir: string, sample: JSCodeSample) {
+    await deps.writeFile(path.join(installDir, INDEX_JS), sample.js, 'utf-8');
+    if (sample.dependencies) {
+      await run(
+          'npm',
+          ['install', '--prefer-offline', '--save'].concat(sample.dependencies),
+          {cwd: installDir});
+    }
+    if (sample.devDependencies) {
+      await run(
+          'npm',
+          ['install', '--prefer-offline', '--save-dev'].concat(
+              sample.devDependencies),
+          {cwd: installDir});
+    }
+    await run('node', [INDEX_JS], {cwd: installDir});
+  }
+}
+
+export function getExistingTarballs(workingDirectory: string) {
+  console.log(`globbing in ${workingDirectory}`);
```

Comment:
Feels like `console.log` would be hard to control inside of the module like this

Reply:
This function is not necessary any more actually.